eng/pipelines/runtime.yml

```diff
@@ -576,6 +594,20 @@ jobs:
 # CoreCLR Test builds using live libraries release build
 # Only when CoreCLR is changed
 #
+- template: /eng/pipelines/common/platform-matrix.yml
```

Comment:
Could you explain why we need to add this pipeline? This seems exactly equal as what you're adding in L633.

Reply:
This is builds the `targetGeneric` tests which were disabled in the template at line L611 by L623.

Reply:
L605 is different than L642. So Not adding this would have resulted in changing what we ran.

Reply:
I guess my question is, why do we need to build the tests against a debug and against a release build of libraries? I need more context on this change, could you elaborate what the requirements are here?

Reply:
My task was to split the building of tests during CI into `targetSpecific` tests and `targetGeneric` tests.  Build the `targetGeneric` tests on OSX.  Continue to run all tests on the platforms where they were currently running.

I think there was an issue here too.  I fixed it in my latest commit.

Reply:
The rationale was we didn't need to build all the tests on all the platforms.  We only needed to build platform specific tests on all platforms.  OSX was chosen because it has historically finished building tests first.

Reply:
Ok, I get that, and why do the generic tests need a release libraries build? That is the part that I don't get.

Reply:
Right now you're adding three jobs. 
1. Libraries build Release
2. CoreCLR build tests (targetGeneric) using a Release libraries build (both on PR and CI).
3. CoreCLR build tests (targetGeneric) using a Debug libraries build (only on PR).

What I don't get why do we need these? As I understand this, we just need to add 1 build here:
1. CoreCLR build tests (targetGeneric) uses Debug on PR and Release on CI libraries.

Reply:
That would change the test coverage.  We do not currently run the set of tests you describe.

The existing YAML looks like this.

```YAML
#
# CoreCLR Test builds using live libraries release build
# Only when CoreCLR is changed
#
- template: /eng/pipelines/common/platform-matrix.yml
  parameters:
    jobTemplate: /eng/pipelines/common/templates/runtimes/build-test-job.yml
    buildConfig: checked
    platforms:
    - Linux_arm 
    - Windows_NT_x86
    - Windows_NT_arm
    - Windows_NT_arm64
    jobParameters:
      testGroup: innerloop
      liveLibrariesBuildConfig: Release
      condition: >-
        or(
          eq(dependencies.checkout.outputs['SetPathVars_coreclr.containsChange'], true),
          eq(variables['isFullMatrix'], true))

#
# CoreCLR Test builds using live libraries debug build
# Only when CoreCLR is changed
#
- template: /eng/pipelines/common/platform-matrix.yml
  parameters:
    jobTemplate: /eng/pipelines/common/templates/runtimes/build-test-job.yml
    buildConfig: checked
    platforms:
    - OSX_x64
    - Linux_x64
    - Linux_arm64
    - Windows_NT_x64
    jobParameters:
      testGroup: innerloop
      liveLibrariesBuildConfig: ${{ variables.debugOnPrReleaseOnRolling }}
      condition: >-
        or(
          eq(dependencies.checkout.outputs['SetPathVars_coreclr.containsChange'], true),
          eq(variables['isFullMatrix'], true))
```

As I read it we always run `Release` version of libraries on a subset of our platforms.

Because I need to build `Debug` and `Release` in some configurations, I need the three jobs.

Reply:
but what does libraries build config matter when you're building the tests? You can build against whatever libraries config, the run-test-job will download libraries again and use that to run, no? 

https://github.com/dotnet/runtime/blob/master/eng/pipelines/common/templates/runtimes/run-test-job.yml#L181

Reply:
I believe you are right.  I hadn't looked closely enough at the run job to realize it didn't matter.

I'll remove the unnecessary legs.

Thanks for explaining the root cause why it wasn't necessary.

It is amazing to me how difficult communication can be 😉

Reply:
No problem. Yeah and specially with so many configurations an templates it’s hard to keep track what is needed and what’s not.

Thanks for elaborating on the reason why you were doing this. Let’s remove the unnecessary extra leg and see if everything goes green which I think it should as run-test-job should download the right libraries build. :fingerscrossed:

Reply:
I pushed the change.